src/providers/mod.rs

```diff
@@ -265,4 +265,14 @@ pub trait Provide {
         trace!("psa_raw_key_agreement ingress");
         Err(ResponseStatus::PsaErrorNotSupported)
     }
+
+    /// Execute a GenerateRandom operation.
+    fn psa_generate_random(
+        &self,
+        _app_name: ApplicationName,
```

Comment:
I'd argue against requesting an `app_name`. The operation can then be performed even for non-authenticated users, I think that should be alright!